internal/authn/oidc/oidc_test.go

```diff
@@ -0,0 +1,200 @@
+package oidc
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"github.com/MicahParks/keyfunc"
+	"github.com/golang-jwt/jwt/v4"
+	"github.com/openfga/openfga/internal/authn"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc/metadata"
+	"log"
+	"strings"
+	"testing"
+)
+
+func TestRemoteOidcAuthenticator_Authenticate(t *testing.T) {
+	t.Run("when_the_authorization_header_is_missing_from_the_gRPC_metadata_of_the_request,_returns_'missing_bearer_token'_error", func(t *testing.T) {
+		authenticator := &RemoteOidcAuthenticator{}
+		_, err := authenticator.Authenticate(context.Background())
+		require.Equal(t, authn.ErrMissingBearerToken, err)
+	})
+	testCases := []struct {
```

Comment:
please add two more test cases
- a token with `exp` (expiration date) in the past should be rejected
- a token with `iat` (issued at) in the future should be rejected